src/diff/children.js

```diff
@@ -111,12 +111,6 @@ export function diffChildren(
 					isHydrating
 				);
 
-				if ((j = childVNode.ref) && oldVNode.ref != j) {
-					if (!refs) refs = [];
```

Comment:
Yay! This array allocation is gone! Now if you attach a ref to a component, we re-use that component's `_renderCallbacks` array. However, if you attach a ref to a DOM VNode, we allocate a new array (that VNode's `_renderCallbacks` array) for that ref. But I think the net is still fewer array allocations.